platform/pocketbook/koreader.app

```diff
@@ -4,6 +4,10 @@ export LC_ALL="en_US.UTF-8"
 # working directory of koreader
 KOREADER_DIR="/mnt/ext1/applications/koreader"
 
+# load our own shared libraries if possible, solely because we don't control InkView, and we'd rather not it have load duplicate system libs...
+# (We handle this via DT_RPATH for our own stuff).
+export LD_LIBRARY_PATH="${KOREADER_DIR}/libs:${LD_LIBRARY_PATH}"
```

Comment:
Unless the InkView library has a RPATH/RUNPATH entry, our executable RPATH entry should still get priority, no?

Reply:
I frankly don't trust it not to do something funky, so I'd rather be vewy, vewy cautious ;).

Reply:
(Plus, I don't know if the base system doesn't have a funky system-wide LD_LIBRARY_PATH, or, worse, an LD_PRELOAD that could do nasty things)

Reply:
And combined with removing the `env -u LD_LIBRARY_PATH ` prefix in `runCommand`?

Reply:
> (Plus, I don't know if the base system doesn't have a funky system-wide LD_LIBRARY_PATH, or, worse, an LD_PRELOAD that could do nasty things)

You could unset those, I think `LD_PRELOAD` still get priority anyway, no?

Reply:
> And combined with removing the `env -u LD_LIBRARY_PATH ` prefix in `runCommand`?

That's done in this PR? Not sure what you were thinking of there ;).


> You could unset those, I think `LD_PRELOAD` still get priority anyway, no?

Yeah, we're can't do anything about a preload hack.

Unsetting LD_LIBRARY_PATH instead could possibly achieve the same thing, but, again, this is a weird platform that I can't test nor trust, so I'm really going brute-force on purpose ;).

Reply:
> >    And combined with removing the env -u LD_LIBRARY_PATH prefix in runCommand?
>
> That's done in this PR? Not sure what you were thinking of there ;).

This change in `frontend/device/sdl/device.lua`:
```diff
 local function runCommand(command)
-     local env = jit.os ~= "OSX" and 'env -u LD_LIBRARY_PATH ' or ""
-     return os.execute(env..command) == 0
+     return os.execute(command) == 0
 end
```

Reply:
Our library directory now takes priority when calling external executable on pocketbook.

Reply:
That's always been the case, because that snippet is an SDL-only codepath, so it never applied to PB.